src/libcore/task/yield_now.rs

```diff
@@ -0,0 +1,60 @@
+use crate::future::Future;
+use crate::pin::Pin;
+use crate::task::{Context, Poll};
+
+/// Cooperatively gives up a timeslice to the executor.
+///
+/// Calling this function calls move the currently executing future to the back
+/// of the execution queue, making room for other futures to execute. This is
+/// especially useful after running CPU-intensive operations inside a future.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// #![feature(task_yield_now)]
+/// # async fn run() {
+/// #
+/// use core::task;
+///
+/// task::yield_now().await;
+/// #
+/// # }
+/// ```
+#[unstable(feature = "task_yield_now", issue = "74331")]
+#[inline]
+pub fn yield_now() -> YieldNow {
+    YieldNow { is_polled: false }
+}
+
+/// Creates a future that yields back to the executor exactly once.
+///
+/// This `struct` is created by the [`yield_now`] function. See its
+/// documentation for more.
+///
+/// [`yield_now`]: fn.yield_now.html
```

Comment:
nit: intra-doc links are my baby ❤️ and since `yield_now` is in scope this should be resolved automatically.
```suggestion
```